### PR TITLE
Allow commentless votes

### PR DESCRIPTION
**Pull request by ThePletch**
First pass at allowing votes to be saved without a comment (and recognizing them as such in Angular).

I'm really new to Angular and coded this without being able to reference the docs, so let me know if I messed up syntax.

### CODE REVIEW

**Comment by ThePletch** [comment]
Resolves #356.

**Comment by kimardenmiller** [comment]
@ThePletch looks good :+1: , though looks as if [your new test is failing](https://travis-ci.org/Spokenvote/spokenvote/builds/64018968).

We don't have to do anything fancy in terms of showing the "no comment" vote as I'll probably reorganize no comment votes into a different area at some point that shows just your profile pic bunched with all the other no comment votes.  But just a note there, in CoffeeScript we try to use "is" where possible, which might not be the case for use since you used "==" and "is" = "===".

Do you have the branch running on your local machine?

**Comment by ThePletch** [comment]
I wrote this without internet access, so I unfortunately was unable to consult the docs on getting this running on my local machine - the failing spec looks like it's just from me relying on a pattern that you don't use, so it should be easy to fix. I'll get things running tonight and fix that up (as well as address the style change).

**Comment by ThePletch** [comment]
Very sorry for how long this took - I spent a much longer time than expected getting postgres set up locally (I'm used to running it in a Vagrant box). The spec should now pass, and I'm replacing `==` with `is` where I can. Do you want me to squash my commits before merging?

**Comment by kimardenmiller** [comment]
Travis appears to be passing. Let me take a look in the morning. Commits are fine.


/k

**Comment by kimardenmiller** [comment]
Code appears to work as expected :+1:  

Regarding the feature in it's usability, here are some thoughts:
 - User tries to vote and sees the screen below. It seems reasonable that she's done, but she's not, as the proposal has not been saved.
image
 - Yes, the fine print says, _"Press Enter to publish your improved version"_, but that's really easy to miss. So we'll probably want to do a vote save with no comment, then use the existing saveEdit method in votes_controller.coffee to save the comment? Other ideas?

Assuming that's what we want to do, I'm easy regarding _who_ does it. If you prefer I do this Angular piece, I'm happy to. I'm also happy to coach you through it.

File: app/assets/javascripts/angular/controllers/votes_controller.coffee [hunk 1]
```diff
@@ -10,7 +10,11 @@ SupportController = [ '$scope', '$location', '$rootScope', 'Vote', ( $scope, $lo
       ($scope.vote
       ), (( response, status, headers, config ) ->
         $rootScope.$broadcast 'event:votesChanged'
-        $scope.alertService.setSuccess 'Your vote was created with the comment: \"' + response.comment + '\"', $scope, 'main'
+        if response.comment is null
+          responseMessage = "Your vote was recorded without a comment"
```
Comment (kimardenmiller) [inline app/assets/javascripts/angular/controllers/votes_controller.coffee]: While your assignment and usage of ```responseMessage``` does work, I think it's technically updated out of scope from your final usage. In any case, my editor complains. The fix would be to insert at line 12: ```responseMessage = undefined```

**Comment by kimardenmiller** [comment]
@ThePletch, a small issue came up, interested in your thoughts.  Unless I prep the saving of a comment-less vote in Angular like so:
```CoffeeScript
    if not $rootScope.sessionSettings.newProposal.votes_attributes or not $rootScope.sessionSettings.newProposal.votes_attributes.comment
      $rootScope.sessionSettings.newProposal.votes_attributes =
        comment: undefined            # Needed for Commentless Voting
```
I get this error:
NoMethodError at /proposals
===========================

> undefined method `merge' for nil:NilClass

app/controllers/proposals_controller.rb, line 63
------------------------------------------------

``` ruby
   58       render json: { editable: proposal.editable?(current_user) }
   59     end
   60   
   61     # POST /proposals.json
   62     def create
>  63       votes_attributes = params[:proposal][:votes_attributes].merge(user_id: current_user.id, ip_address: request.remote_ip)
   64       @proposal = current_user.proposals.create(proposal_params)
   65   
   66       if @proposal.new_record?
   67         render json: @proposal.errors, status: :unprocessable_entity
   68       else
```

App backtrace
-------------

 - app/controllers/proposals_controller.rb:63:in `create'



Seem like Rails should be the one to handle it, or is my Angular fix the right way to go?

**Comment by ThePletch** [comment]
Hey Kim, looks like you just need to add a null guard there (in case that
parameters field isn't provided). That's definitely a rails level task. I
can do it tonight after fireworks in Boston.

**Comment by kimardenmiller** [comment]
Awesome, Steve :-)




Just starting roof-top cookout, SF time.


/k

**Pull request merged**